funcframework/framework_test.go

```diff
@@ -632,6 +632,17 @@ func TestDeclarativeFunctionCloudEvent(t *testing.T) {
 	}
 }
 
+func TestDeclarativeFunctionWithoutRegister(t *testing.T) {
+	funcName := "HelloWorld"
+	os.Setenv("FUNCTION_TARGET", funcName)
+
+	wantErr := "no matching function found"
+
+	if err := Start("4444"); err.Error() != wantErr {
```

Comment:
Apparently, there's a built-in golang way to get a random, open port: https://stackoverflow.com/questions/43424787/how-to-use-next-available-port-in-http-listenandserve